### PR TITLE
Update dependecies for PHP 8.0 compatibility

### PR DESCRIPTION
**Pull request by ausi**
See #2263 

~~I think we can remove `phpunit/php-token-stream`? 
Or do we need this package anywhere in the core?~~

### CODE REVIEW

**Comment by ausi** [comment]
We are using it in the `Contao\CoreBundle\Config\Loader\PhpFileLoader`

**Comment by m-vo** [comment]
> We are using it in the Contao\CoreBundle\Config\Loader\PhpFileLoader

We should probably replace it at some point, though, because this package is abandoned.

**Comment by leofeyer** [comment]
Can we maybe rework the class so it works without the package? Maybe using the PHP function `token_get_all()`?

**Comment by m-vo** [comment]
That's what `PHP_Token_Stream` is doing as well: https://github.com/sebastianbergmann/php-token-stream/blob/master/src/Stream.php#L413-L514

Maybe this function gets small enough if we reduce it to our use case to adopt it.

**Comment by ausi** [comment]
I think this would be quite some work to do. If we want to get rid of the package, we should create a seperate PR for that.

**Comment by ausi** [comment]
@bytehead can you confirm that `spomky-labs/otphp` 10.0 is compatible with our code?

**Comment by bytehead** [comment]
I'll check and let you know!

**Comment by bytehead** [comment]
Looks good so far: 

> For the upgrade from v9.x to v10.x, there is normally nothing to do. The main changes only concern:
>
>
> - new public methods added to the interfaces.
> - a new interface for the factory.
> - the parameters of the method OTP::getQrCodeUri are now required as the Google API is deprecated since April 2019.
>
>The library now relies on the MBString extension. This extension is now a required dependency.

https://github.com/Spomky-Labs/otphp/blob/v10.0/doc/UPGRADE_v9-v10.md

**Comment by m-vo** [comment]
> Can we maybe rework the class so it works without the package? Maybe using the PHP function token_get_all()?

I had a quick look at it. Parsing and stripping PHP tags is the easy part. Unfortunately we would have to write/adapt quite a lot of code to handle all the namespace edge cases, which makes it - well - a doable but non-trivial task.

**Comment by leofeyer** [comment]
Maybe we can use https://github.com/nikic/PHP-Parser?

**Comment by ausi** [comment]
> Maybe we can use [nikic/PHP-Parser](https://github.com/nikic/PHP-Parser)?

See #2282

File: composer.json [hunk 1]
```diff
@@ -70,13 +70,13 @@
         "paragonie/constant_time_encoding": "^2.2",
         "patchwork/utf8": "^1.2",
         "phpspec/php-diff": "^1.0",
-        "phpunit/php-token-stream": "^1.4 || ^2.0 || ^3.0",
+        "phpunit/php-token-stream": "^1.4 || ^2.0 || ^3.0 || ^4.0",
```
Comment (leofeyer) [inline composer.json:73]: Do we really still need version 1 and 2?

Comment (ausi) [reply composer.json:73]: Yes, bugfix versions should not require higher major dependency versions IMO (unless necessary).
If we don’t need versions 1 and 2 we should remove that in 4.11 I think.

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @ausi.

**Pull request merged**